bundles/org.openhab.binding.sonnen/src/main/java/org/openhab/binding/sonnen/internal/SonnenHandler.java

```diff
@@ -0,0 +1,276 @@
+            validConfig = false;
+        }
+        if (config.hostIP == null) {
+            errors += " Parameter 'hostIP' must be configured.";
+            statusDescr = "IP Address must be configured!";
+            validConfig = false;
+        }
+        errors = errors.trim();
+        Helper message = new Helper();
+        message.setStatusDescription(statusDescr);
+        if (validConfig) {
+            serviceCommunication.setConfig(config);
+            if (serviceCommunication.refreshBatteryConnection(message, this.getThing().getUID().toString())) {
+                updateStatus(ThingStatus.ONLINE);
+                updateLinkedChannels();
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, message.getStatusDesription());
+            }
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, message.getStatusDesription());
+        }
+    }
+
+    /**
+     * Calls the service to update the battery data
+     *
+     * @param postdata
+     */
+    private boolean updatebatteryData(@Nullable String postdata) {
+        Helper message = new Helper();
+        if (serviceCommunication.refreshBatteryConnection(message, this.getThing().getUID().toString())) {
+            updateStatus(ThingStatus.ONLINE);
+            return true;
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                    message.getStatusDesription());
+            return false;
+        }
+    }
+
+    private void updateLinkedChannels() {
+        verifyLinkedChannel(CHANNELBATTERYCHARGING);
+        verifyLinkedChannel(CHANNELBATTERYDISCHARGING);
+        verifyLinkedChannel(CHANNELGRIDFEEDIN);
+        verifyLinkedChannel(CHANNELCONSUMPTION);
+        verifyLinkedChannel(CHANNELSOLARPRODUCTION);
+        verifyLinkedChannel(CHANNELBATTERYLEVEL);
+        verifyLinkedChannel(CHANNELFLOWCONSUMPTIONBATTERY);
+        verifyLinkedChannel(CHANNELFLOWCONSUMPTIONGRID);
+        verifyLinkedChannel(CHANNELFLOWCONSUMPTIONPRODUCTION);
+        verifyLinkedChannel(CHANNELFLOWGRIDBATTERY);
+        verifyLinkedChannel(CHANNELFLOWPRODUCTIONBATTERY);
+        verifyLinkedChannel(CHANNELFLOWPRODUCTIONGRID);
+
+        if (!linkedChannels.isEmpty()) {
+            updatebatteryData(null);
+            for (Channel channel : getThing().getChannels()) {
+                updateChannel(channel.getUID().getId());
+            }
+            startAutomaticRefresh();
+            automaticRefreshing = true;
+        }
+    }
+
+    private void verifyLinkedChannel(String channelID) {
+        if (isLinked(channelID) && !linkedChannels.containsKey(channelID)) {
+            linkedChannels.put(channelID, true);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        stopScheduler();
+    }
+
+    private void stopScheduler() {
```

Comment:
`stopAutomaticRefresh` could be a better name to show that it is the reverse of `startAutomaticRefresh`

Reply:
ok